test/elixir_internal_certificate_web/controllers/api/v1/user_search_controller_test.exs

```diff
@@ -152,4 +172,103 @@ defmodule ElixirInternalCertificateWeb.Api.V1.UserSearchControllerTest do
              }
     end
   end
+
+  describe "GET show/2" do
+    test "when logged in user, given a valid user_search id, returns result and 200 status", %{
+      conn: conn
+    } do
+      user = insert(:user)
+      user_search = insert(:user_search, status: :success, user: user, id: 1)
+
+      %{
+        ad_words_total: ad_words_total,
+        html_response: html_response,
+        links_total: links_total,
+        non_ad_words_links: non_ad_words_links,
+        non_ad_words_total: non_ad_words_total,
+        top_ad_words_links: top_ad_words_links,
+        top_ad_words_total: top_ad_words_total
+      } =
+        _search_result =
+        insert(
+          :search_result,
+          user_search: user_search,
+          id: 3
+        )
+
+      another_user_keyword = insert(:user_search, id: 2)
+      _another_search_result = insert(:search_result, user_search: another_user_keyword, id: 4)
+
+      conn =
+        conn
+        |> token_auth_user(user)
+        |> get(Routes.api_user_search_path(conn, :show, 1))
+
+      assert %{
+               "data" => %{
+                 "attributes" => %{
+                   "ad_words_total" => ^ad_words_total,
+                   "html_response" => ^html_response,
+                   "links_total" => ^links_total,
+                   "non_ad_words_links" => ^non_ad_words_links,
+                   "non_ad_words_total" => ^non_ad_words_total,
+                   "top_ad_words_links" => ^top_ad_words_links,
+                   "top_ad_words_total" => ^top_ad_words_total
+                 },
+                 "id" => "3",
+                 "relationships" => %{},
+                 "type" => "search_result"
+               },
+               "included" => []
+             } = json_response(conn, 200)
+    end
+
+    test "when logged in user, given a not existing user_search id, returns 404 status",
```

Comment:
we are missing a case

1. Existing user search id but that doesn't belongs to the logged in user

Reply:
Added in [2cf1b4d](https://github.com/nvminhtue/elixir_internal_certificate/pull/45/commits/2cf1b4de605fa19c371ddfbf8efce1d048adbf7c)